src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -154,6 +158,7 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
             // Take the package from the first source it is found in
             foreach (var source in sources)
             {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This one seems a bit excessive right? 
You are passing the cancellation token to each of the `GetPackageMetadataFromLocalSourceAsync` methods which in turn calls GetMetadataAsync to local sources: 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/LocalRepositories/FindLocalPackagesResourceV2.cs
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/LocalRepositories/FindLocalPackagesResourceV3.cs 

Both of which take the token and respect it.

Reply:
I don't know which sources honor the token, and which don't. I like the idea that I stop iterating over potentially slow sources as soon as I am aware of it. 

I didn't realize that we are guaranteed to use only 2 possible Resource implementations... so if that's true, yes I'll agree it's excessive and remove it.

Reply:
It's up to the implementation to determine when it's safe to respect the token. 

In this case the method is `GetLocalPackageMetadataAsync`, and that one will only lead to those 2 different implementations.